src/components/Pages/WithText/Buttons/ToggleScan/ScanPDF.js

```diff
@@ -0,0 +1,112 @@
+import React, { useState, useEffect, useRef, useCallback, useContext } from 'react';
+import { useLocation, useNavigate } from 'react-router-dom';
+import { getQuery, stringify } from '../../../../../helpers/url';
+import { Button } from 'semantic-ui-react';
+import { Document, Page, pdfjs } from 'react-pdf';
+import { ErrorSplash, LoadingSplash } from '../../../../shared/Splash/Splash';
+import { physicalFile } from '../../../../../helpers/utils';
+import { useSelector } from 'react-redux';
+import { textPageGetScanFileSelector } from '../../../../../redux/selectors';
+import { useTranslation } from 'react-i18next';
+import ScanToolbar from './ScanToolbar';
+import { ScanZoomContext } from './ScanZoomContext';
+
+const ScanPDF = () => {
+  const { t }              = useTranslation();
+  const location           = useLocation();
+  const navigate           = useNavigate();
+  const { scan, ...query } = getQuery(location);
+  const file               = useSelector(textPageGetScanFileSelector);
+  const ref                = useRef(null);
+
+  const [width, setWidth]           = useState();
+  const [numPages, setNumPages]     = useState();
+  const [inputValue, setInputValue] = useState(1);
+  const { zoom }                    = useContext(ScanZoomContext);
+
+  useEffect(() => {
+    if (!ref.current) {
+      return null;
+    }
+
+    const setDivSize = () => setWidth(ref.current.getBoundingClientRect().width);
+    setDivSize();
+    window.addEventListener('resize', setDivSize);
+    return () => {
+      window.removeEventListener('resize', setDivSize);
+    };
+  }, [file]);
```

Comment:
How does this useEffect depends on `file`?

Reply:
fixed